cmd/snap/cmd_run_test.go

```diff
@@ -226,6 +244,9 @@ func (s *RunSuite) TestSnapRunAppRunsChecksInhibitionLock(c *check.C) {
 	var execArg0 string
 	var execArgs []string
 	restorer := snaprun.MockSyscallExec(func(arg0 string, args []string, envv []string) error {
+		// lock should be released before calling snap-confine using beforeExec() callback
```

Comment:
I completely agree with you, it currently has a single purpose and it makes sense to name it accordingly. I was aiming for something generic because it is being used also in `snapRunHook` and the callback now either closes the flock or triggers a retry so the logic can grow.

Reply:
Yes, with the retry, I think the generic name is better.